docs/user-guide/kubernetes.md

```diff
@@ -330,6 +330,62 @@ echo "$(minikube ip) traefik-ui.minikube" | sudo tee -a /etc/hosts
 
 We should now be able to visit [traefik-ui.minikube](http://traefik-ui.minikube) in the browser and view the Træfik Web UI.
 
+## Basic Authentication
+
+It's possible to add additional authentication annotations in the Ingress rule. The source of the authentication is a secret that contains usernames and passwords inside the the key auth. To read about basic auth limitations see the [Kubernetes Ingress](/configuration/backends/kubernetes) configuration page. 
+
+#### Creating the Secret
+
+1. Use `htpasswd` to create a file containing the username and the base64-encoded password:
+
+   `htpasswd -c ./auth myusername`  
+
+   You will be prompted for a password which you will have to enter twice. htpasswd will create
+   a file with the following:  
+
+   ```
+   ~$ cat auth  
+
+   myusername:$apr1$78Jyn/1K$ERHKVRPPlzAX8eBtLuvRZ0
+   ```  
+
+2. Now use `kubectl` to create a secret in the monitoring namespace using the file created by `htpasswd`.
+
+   `kubectl create secret generic mysecret --from-file auth --namespace=monitoring`
+
+   **Note:** Secret must be in same namespace as the ingress rule.
+
+3. Create the ingress using the following annotations to specify basic auth and that the username and password is stored in "mysecret".  
+
+   - `ingress.kubernetes.io/auth-type`: `"basic"`
+   - `ingress.kubernetes.io/auth-secret`: `"mysecret"`
+
+   Following is a full ingress example based on Prometheus:  
+
+   ```yaml
```

Comment:
Could you replace `yaml` by `yml` ?

Reply:
@ldez FWIW, except for one case, the guide uses `yaml` everywhere else.